rust/benches/caching.rs

```diff
@@ -0,0 +1,321 @@
+use std::{
+    self,
+    hash::RandomState,
+    sync::{
+        Arc,
+        atomic::{AtomicU16, AtomicU64, Ordering},
+    },
+};
+
+use carmen_rust::{
+    error::BTResult,
+    node_manager::{
+        NodeManager,
+        cached_node_manager::CachedNodeManager,
+        lock_cache::{self, EvictionHooks},
+    },
+    storage::{self, Storage},
+};
+use criterion::{BenchmarkId, criterion_group, criterion_main};
+use quick_cache::{Lifecycle, UnitWeighter};
+
+use crate::utils::{execute_with_threads, pow_2_threads, with_prob};
+pub mod utils;
+
+/// Lifecycle implementation that randomly pins items based on a given probability
+#[derive(Clone, Default)]
+pub struct RandomLifecycle {
+    prob: u16,
+}
```

Comment:
```suggestion
/// Lifecycle implementation that randomly pins items based on a given probability
#[derive(Clone, Default)]
pub struct RandomPinner {
    prob: u16,
}
```

You could have a single type that implements both `Lifecycle` and `EvictionHooks` (i.e., get rid of `ProbPinnedHook`).